### DISCUSSION TITLE
how can i change the database url

### DISCUSSION THREAD

**Discussion by bytesagar** [prisma/prisma#10500] [Q&A] [answered]
So i have a postgres database running locally and one on aws cloud.
I am facing error like can't connect to the database and when i tried
```
?sslmode=disable&connect_timeout=100&pool_timeout=100&socket_timeout=100
```
It threw error like
```
Error: P1010: User `puzvtucmxjlilz` was denied access on the database `d4jk6is9k6nftd.public`
```
So what i wanted to ask is I have to manually change the databse url in prisma.schema file.
Is there any way i can set it automatically. while running or pushing to github

**Comment by austincrim** [comment, marked as answer]
Hey @bytesagar,

The way to handle this is through **environment variables**. Specify your connection string in the `schema.prisma` like so: 
```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

and it will automatically look for a `DATABASE_URL` environment variable. Out of the box, Prisma will also load a `.env` file if it is in your project folder.
```.env
DATABASE_URL="connection string here"
```

Loading secrets or environment variables in AWS can take many different forms depending on your setup, but each relevant AWS service should have documentation on that.

Let me know if you have questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
